docs/root/configuration/http/http_filters/fault_filter.rst

```diff
@@ -42,6 +42,12 @@ x-envoy-fault-abort-request
   In order for the header to work, :ref:`header_abort
   <envoy_api_field_config.filter.http.fault.v2.FaultAbort.header_abort>` needs to be set.
 
+x-envoy-fault-abort-grpc-request
```

Comment:
Yes,  I think that we should add a support for `x-envoy-fault-abort-grpc-request-percentage` HTTP header (whose implementation would be based of how `x-envoy-fault-abort-request-percentage` HTTP header is defined).

Together with @buildbreaker @murki we discussed potential scenarios where having `x-envoy-fault-abort-grpc-request-percentage` HTTP header would be helpful to test client-side implementation of some of the GRPC requests that Lyft mobile applications perform. This lets me believe that the addition of such header will be useful for engineers from other companies too.

Reply:
Sorry I missed one thing when I posted my initial comment here. 

Initially, I thought that GRPC aborts would be applied on top of http aborts meaning something like:

1. Requests comes in
2. Check whether to abort the request with http fault - look at `x-envoy-fault-abort-request-percentage` and `x-envoy-fault-abort-request` HTTP headers. 
3. If no http fault was applied (because percentage was <100% and we decided not to apply HTTP fault) check whether grpc faults should be applied - look at `x-envoy-fault-abort-grpc-request-percentage` and `x-envoy-fault-abort-grpc-request` HTTP headers. 

and for this reason I thought that we needed the addition of `x-envoy-fault-abort-grpc-request-percentage` HTTP header. 

Now that I know that the implementation looks more like apply "either HTTP or GRPC abort" (only one of them) I don't see any benefits of us having `x-envoy-fault-abort-grpc-request-percentage` HTTP header. Don't get me wrong, as far as I can tell current implementation works correctly and it unlocks GRPC abort tests for mobile people at Lyft but the addition of `x-envoy-fault-abort-grpc-request-percentage` HTTP header itself doesn't add any value in here - we could continue to use `x-envoy-fault-abort-request-percentage` HTTP request header and functionality-wise we wouldn't lose anything. We could keep it but I think that removal makes a little more sense since it would keep the implementation simpler.

I really feel bad asking you to do this @vijayrajput1 but could we get rid of `x-envoy-fault-abort-grpc-request-percentage` HTTP header and use `x-envoy-fault-abort-request-percentage` for both HTTP and GRPC aborts? My deep apologizes here - I know that the addition of this header costed you a significant amount of work in the first place so it really feels bad to ask you to revert some of your changes.

Reply:
Thank you @Augustyniak , I actually had this (using one header) approach in mind earlier but then I kinda convinced myself having a matching header would make things more clear. Meaning use `x-envoy-fault-abort-grpc-request-percentage` for `x-envoy-fault-abort-grpc-request` vs use `x-envoy-fault-abort-request-percentage` for `x-envoy-fault-abort-grpc-request`. I should have raised this point earlier. I'm okay, with either of those approaches - my take is if having a separate header `x-envoy-fault-abort-grpc-request-percentage` isn't making things anymore clear than we should use only one header i.e `x-envoy-fault-abort-request-percentage` - @mattklein123 any preference ?

Reply:
Yeah agreed on using 1 header. Sorry, I didn't look very deeply at this before punting review over to @Augustyniak. Apologies also!